core/dbt/include/global_project/macros/materializations/models/materialized_view/materialized_view.sql

```diff
@@ -0,0 +1,121 @@
+
+    {{ _setup(backup_relation, intermediate_relation, pre_hooks) }}
+
+        {% set build_sql = _get_build_sql(existing_relation, target_relation, backup_relation, intermediate_relation) %}
+
+        {% if build_sql == '' %}
+            {{ _execute_no_op(target_relation) }}
+        {% else %}
+            {{ _execute_build_sql(build_sql, existing_relation, target_relation, post_hooks) }}
+        {% endif %}
+
+    {{ _teardown(backup_relation, intermediate_relation, post_hooks) }}
+
+    {{ return({'relations': [target_relation]}) }}
+
+{% endmaterialization %}
+
+
+{% macro _setup(backup_relation, intermediate_relation, pre_hooks) %}
+
+    -- backup_relation and intermediate_relation should not already exist in the database
+    -- it's possible these exist because of a previous run that exited unexpectedly
+    {% set preexisting_backup_relation = load_cached_relation(backup_relation) %}
+    {% set preexisting_intermediate_relation = load_cached_relation(intermediate_relation) %}
+
+    -- drop the temp relations if they exist already in the database
+    {{ drop_relation_if_exists(preexisting_backup_relation) }}
+    {{ drop_relation_if_exists(preexisting_intermediate_relation) }}
+
+    {{ run_hooks(pre_hooks, inside_transaction=False) }}
+
+{% endmacro %}
+
+
+{% macro _teardown(backup_relation, intermediate_relation, post_hooks) %}
+
+    -- drop the temp relations if they exist to leave the database clean for the next run
+    {{ drop_relation_if_exists(backup_relation) }}
+    {{ drop_relation_if_exists(intermediate_relation) }}
+
+    {{ run_hooks(post_hooks, inside_transaction=False) }}
+
+{% endmacro %}
+
+
+{% macro _get_build_sql(existing_relation, target_relation, backup_relation, intermediate_relation) %}
+
+    {% set full_refresh_mode = should_full_refresh() %}
+
+    -- determine the scenario we're in: create, full_refresh, alter, refresh data
+    {% if existing_relation is none %}
+        {% set build_sql = get_create_materialized_view_as_sql(target_relation, sql) %}
+    {% elif full_refresh_mode or not existing_relation.is_materialized_view %}
+        {% set build_sql = get_replace_materialized_view_as_sql(target_relation, sql, existing_relation, backup_relation, intermediate_relation) %}
+    {% else %}
+
+        -- get config options
+        {% set on_configuration_change = config.get('on_configuration_change') %}
+        {% set configuration_changes = get_materialized_view_configuration_changes(existing_relation, config) %}
+
+        {% if configuration_changes == {} %}
+            {% set build_sql = refresh_materialized_view(target_relation) %}
+
+        {% elif on_configuration_change == 'apply' %}
+            {% set build_sql = get_alter_materialized_view_as_sql(target_relation, configuration_changes, sql, existing_relation, backup_relation, intermediate_relation) %}
+        {% elif on_configuration_change == 'skip' %}
```

Comment:
Make this a RunStatus.Skipped

Reply:
I cannot figure out how to update the NodeStatus from within the materialization. It appears that the jinja context only has Relation and Config level attributes at run (i.e. we're already "inside" the node). And the runner either captures that an exception was thrown, or not. Seed is able to work because it's a separate command on the CLI, and get's dispatched to its own task and runner. We don't want to do that here since we want this to behave like any other model materialization (e.g. table, view).

Reply:
We've decided to leave this is a RunStatus.Success, since the node did finish running and should not impede downstream models from running. However, we updated the `on_configuration_change` setting to be "continue" to avoid confusion.